resource-managers/kubernetes/core/src/test/scala/org/apache/spark/deploy/k8s/submit/KubernetesDriverBuilderSuite.scala

```diff
@@ -89,6 +97,29 @@ class KubernetesDriverBuilderSuite extends SparkFunSuite {
       SECRETS_STEP_TYPE)
   }
 
+  test("Apply Python step if main resource is python.") {
+    val conf = KubernetesConf(
```

Comment:
Depends on what we want to use the builder for. One advantage of a builder is handled by case classes already: the fact that you don't have to order arguments in a particular way; you can get around this by using named parameters when you construct the object. But, if you want to stage the construction of the object in multiple calls, then a builder will get you that while a case class by itself will not.

I think it would be neater to have a builder. The [SparkSession Builder](https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala#L775) is an example from the project we can follow.